### DISCUSSION TITLE
Example site does not start up

### DISCUSSION THREAD

**Discussion by deining** [gethinode/hinode#768] [Help] [answered]
**Describe the bug**

I cloned the repo and tried to run the example site. Hugo is downloading modules for a long, long time, seems like I'm encountering infinite recursion here.

**To reproduce**
1. Clone repo
2. Change to dir èxampleSite`
4. Execute `hugo`
5. See infinite recursion

**Expected behavior**
Example site should start up.

**Log file**

I removed all module definitions from `go.mod` of example site. Afterwards, I'm seeing:

~/hinode/exampleSite$ hugo 
go: added github.com/gethinode/mod-bootstrap v1.2.2
go: added github.com/gethinode/mod-flexsearch v1.9.0
go: added github.com/gethinode/mod-fontawesome v1.8.1
go: added github.com/gethinode/mod-katex v1.0.5
go: added github.com/gethinode/mod-leaflet v1.0.0
hugo: downloading modules …
go: added github.com/gethinode/mod-lottie v1.4.2
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
hugo: downloading modules …
^C

**Environment (please complete the following information):**
 - Hinode version: HEAD at `aa5e6c6`
 - Host OS: [Ubuntu 22.04]
 - Node version: node v20.11.0]
 - Hugo version: hugo v0.122.0-b9a03bd59d5f71a529acb3e33f995e0ef332b3aa+extended linux/amd64
 - Browser: n/a

**Comment by markdumay** [comment]
The example site uses a module replacement to source the files locally, see `exampleSite/config/_default/hugo.toml`:

```
[module]
  replacements = 'github.com/gethinode/hinode -> ../..'
  [[module.imports]]
    path = "github.com/gethinode/hinode"
```

As such, your commands enter an infinite loop. Try running the following from your **repo root** instead:

```bash
hugo mod vendor
hugo -s exampleSite
```

**Comment by deining** [comment, marked as answer]
> Try running the following from your **repo root** instead:
> 
> ```shell
> hugo mod vendor
> hugo -s exampleSite
> ```

This works, thanks for helping me out here.
I don't find this behavior intuitive, though. I authored #769 to make running the example site more standard conform and accessible.
